entries/QUnit.module.xml

```diff
@@ -15,16 +15,34 @@
 				<desc>Runs after each test</desc>
 			</property>
 		</argument>
+		<argument name="nested" optional="true">
+			<desc>A callback with grouped tests and nested modules to run under the current module label.</desc>
+			<type name="Function">
+				<argument name="hooks" type="Object"></argument>
+			</type>
+		</argument>
 	</signature>
 	<desc>
 		Group related tests under a single label.
 	</desc>
 	<longdesc>
 		<p>
-			All tests that occur after a call to <code>QUnit.module()</code> will be grouped into that module, until other <code>QUnit.module()</code> is called. The test names will all be preceded by the module name in the test results. You can then use that module name to select tests to run.
+			You can then use the module name to organize, select and filter tests to run.
+		</p>
+		<p>
+			All tests inside a module callback function will be grouped into that module. The test names will all be preceded by the module name in the test results. Other modules can be nested inside this callback function, where their tests names will be labeled by their names recursively prefixed by their parent modules.
+		</p>
+		<p>
+			If <code>QUnit.module</code> is defined without a <code>nested</code> callback argument, all subsequently defined tests will be grouped into the module until another module is defined.
+		</p>
+		<p>
+			Modules with grouped tests functions allow defining nested modules, and QUnit will run tests on the parent module before going deep on the nested ones, even if they're declared first. <code>beforeEach</code> and <code>afterEach</code> callbacks on a nested module call will stack to the parent hooks.
 		</p>
 		<p>
-			You can specify code to run before and after each test using the hooks argument, and also to create properties that will be shared on the context of each test. Any additional properties on the hooks object will be added to that context. The callbacks can also be used to add properties.
+			You can specify code to run before and after each test using the hooks argument, and also to create properties that will be shared on the context of each test. Any additional properties on the hooks object will be added to that context. The hooks parameter is argument is still optional if you call <code>QUnit.module</code> with a callback argument.
+		</p>
+		<p>
+			The module callback is invoked with the test environment as its <code>this</code> context, which properties are shared to its tests, hooks and nested modules. Note that changes on tests <code>this</code> won't affect sibling tests, where <code>this</code> will be reset to the same value for each test.
```

Comment:
@jzaefferer agreed it sounds awkward. Maybe recasting the sentence would work better. Something like this?…

"The module callback is invoked with the test environment as its <code>this</code> context, with the environment's properties shared to the module's tests, hooks, and nested modules."
(Note that I also added a comma after "hooks" because I'm pretty sure we've standardized on using the Oxford comma.)

Since I'm reading this out of context, I'm still seeing a couple ambiguities, so ignore these warning flags if you think they're obvious to someone who is grokking all of this:

* It is the test environment's properties that are shared, correct? 
* And I'm not sure what "its tests" is referring to: is it the module's tests or or the module callback's tests? I mean, I would assume it's the module's tests, but the module itself is never referenced in that sentence.
* I'm not familiar with the construction, "shared to." Not sure if that's a British variant of "shared with" (as "different to" is of "different from") or if it's carrying some specific meaning.